src/runtime/propensity-server.js

```diff
@@ -238,18 +239,71 @@ export class PropensityServer {
       method: 'GET',
       credentials: 'include',
     });
-    const url =
-      adsUrl('/subopt/pts?products=') +
-      this.publicationId_ +
-      '&type=' +
-      type +
-      '&ref=' +
-      referrer;
+    const pubId = this.deps_.pageConfig().getPublicationId();
+    let url = addQueryParam(adsUrl('/subopt/pts'), 'products', pubId);
+    url = addQueryParam(url, 'type', type);
+    url = addQueryParam(url, 'ref', referrer);
     return this.fetcher_
       .fetch(this.propensityUrl_(url), init)
       .then(result => result.json())
       .then(response => {
         return this.parsePropensityResponse_(response);
       });
   }
+
+  /**
+   * @param {string} url
+   * @private
+   */
+  fetchWithGdpr_(url) {
+    const init = /** @type {!../utils/xhr.FetchInitDef} */ ({
+      method: 'GET',
+      credentials: 'include',
+    });
+    url = this.propensityUrl_(url);
+    this.validateGDPR_();
+    if (this.consentStringPromise_ === null) {
+      return this.fetcher_.fetch(url, init);
+    }
+    return this.consentStringPromise_.then(tcdata => {
+      url = addQueryParam(url, 'gdpr_consent', tcdata.tcString || '');
+      url = addQueryParam(url, 'gdpr', tcdata.gdprApplies ? '1' : '0');
+      return this.fetcher_.fetch(url, init);
+    });
+  }
+
+  /**
+   * Calling this function ensures GDPR consent is obtained prior to submitting
+   * data to Propensity as long as the gdprVendorIds are set in config.
+   * @private
+   */
+  validateGDPR_() {
+    const vendors = this.deps_.config().gdprVendorIds;
+    if (!vendors) {
+      this.vendors_ = null;
+      this.consentStringPromise_ = null;
+      this.gdpr_ = null;
+      return;
+    }
+    if (
+      this.consentStringPromise_ !== null &&
+      doArraysMatch(vendors, this.vendors_)
+    ) {
+      return;
+    }
+    this.vendors_ = vendors;
+    this.gdpr_ = new TcfV2(this.win_);
+    let resolver;
+    this.consentStringPromise_ = new Promise(
+      resolve => void (resolver = resolve)
+    );
+    const callback = TCData => {
+      if (TCData.tcString === TC_STRING_UNAVAILABLE) {
```

Comment:
Note here that I am assuming "unavailable" is temporary and not permanent.  So it loops until it works.  If this is not a valid assumption let me know.